src/sentry/explore/translation/alerts_translation.py

```diff
@@ -0,0 +1,211 @@
+from sentry.workflow_engine.models.data_condition import DataCondition
+from sentry.workflow_engine.models.data_source import DataSource
+
+logger = logging.getLogger(__name__)
+
+
+def snapshot_snuba_query(snuba_query: SnubaQuery):
+    if snuba_query.dataset in [Dataset.PerformanceMetrics.value, Dataset.Transactions.value]:
+        query_snapshot = {
+            "type": snuba_query.type,
+            "dataset": snuba_query.dataset,
+            "query": snuba_query.query,
+            "aggregate": snuba_query.aggregate,
+            "time_window": snuba_query.time_window,
+        }
+        snuba_query.query_snapshot = query_snapshot
+        snuba_query.save()
+
+    return snuba_query
+
+
+def _get_old_query_info(snuba_query: SnubaQuery):
+    old_query_type = SnubaQuery.Type(snuba_query.type)
+    old_dataset = Dataset(snuba_query.dataset)
+    old_query = snuba_query.query
+    old_aggregate = snuba_query.aggregate
+
+    return old_query_type, old_dataset, old_query, old_aggregate
+
+
+def translate_detector_and_update_subscription_in_snuba(snuba_query: SnubaQuery):
+    data_source: DataSource = DataSource.objects.get(
+        source_id=str(snuba_query.id), type=DATA_SOURCE_SNUBA_QUERY_SUBSCRIPTION
+    )
+    if not features.has(
+        "organizations:migrate-transaction-alerts-to-spans", data_source.organization
+    ):
+        logger.info("Feature flag not enabled")
+        return
+
+    detectors = data_source.detectors.all()
+    snapshot_snuba_query(snuba_query)
+
+    snapshot = snuba_query.query_snapshot
+    if not snapshot:
+        return
+
+    old_query_type, old_dataset, old_query, old_aggregate = _get_old_query_info(snuba_query)
+
+    eap_query_parts, dropped_fields = translate_mep_to_eap(
+        QueryParts(
+            selected_columns=[snapshot["aggregate"]],
+            query=snapshot["query"],
+            equations=None,
+            orderby=None,
+        )
+    )
+
+    if dropped_fields["selected_columns"]:
+        return
+
+    translated_aggregate = eap_query_parts["selected_columns"][0]
+    translated_query = eap_query_parts["query"]
+
+    snuba_query.aggregate = translated_aggregate
+    snuba_query.query = translated_query
+    snuba_query.dataset = Dataset.EventsAnalyticsPlatform.value
+
+    if snapshot["dataset"] == Dataset.PerformanceMetrics.value:
+        snuba_query.extrapolation_mode = ExtrapolationMode.SERVER_WEIGHTED.value
+    elif snapshot["dataset"] == Dataset.Transactions.value:
+        snuba_query.extrapolation_mode = ExtrapolationMode.NONE.value
+
+    with atomic_transaction(
+        using=(
+            router.db_for_write(SnubaQuery),
+            router.db_for_write(SnubaQueryEventType),
+        )
+    ):
+        snuba_query.save()
+        SnubaQueryEventType.objects.filter(snuba_query=snuba_query).delete()
+        SnubaQueryEventType.objects.create(
+            snuba_query=snuba_query, type=SnubaQueryEventType.EventType.TRACE_ITEM_SPAN.value
+        )
```

Comment:
Should we just update this to the new type instead? Or are you concerned there could be multiple event types here?